--- ./javaparser-core/target/generated-sources/javacc/com/github/javaparser/ASTParser.java.orig	2017-02-23 12:06:56.689597478 +0100
+++ ./javaparser-core/target/generated-sources/javacc/com/github/javaparser/ASTParser.java	2017-02-23 12:07:09.593598012 +0100
@@ -179,15 +179,17 @@
     }
 
     static final class GTToken extends Token {
+        final String originalImage;
         int realKind = ASTParserConstants.GT;
 
-        GTToken(int kind, String image) {
+        GTToken(int kind, String image, String originalImage) {
+            this.originalImage = originalImage;
             this.kind = kind;
             this.image = image;
         }
 
-        public static Token newToken(int kind, String image) {
-            return new GTToken(kind, image);
+        public static Token newToken(int kind, String image, String originalImage) {
+            return new GTToken(kind, image, originalImage);
         }
     }
 
@@ -8921,7 +8923,7 @@
 
   /** Generated Token Manager. */
   public ASTParserTokenManager token_source;
-  JavaCharStream jj_input_stream;
+  JavaCharStream2 jj_input_stream;
   /** Current token. */
   public Token token;
   /** Next token. */
@@ -8967,7 +8969,7 @@
 
   /** Constructor. */
   public ASTParser(Provider stream) {
-    jj_input_stream = new JavaCharStream(stream, 1, 1);
+    jj_input_stream = new JavaCharStream2(stream, 1, 1);
     token_source = new ASTParserTokenManager(jj_input_stream);
     token = new Token();
     jj_ntk = -1;
@@ -8987,7 +8989,7 @@
   /** Reinitialise. */
   public void ReInit(Provider stream) {
 	if (jj_input_stream == null) {
-      jj_input_stream = new JavaCharStream(stream, 1, 1);
+      jj_input_stream = new JavaCharStream2(stream, 1, 1);
    } else {
       jj_input_stream.ReInit(stream, 1, 1);
    }
